spec/dynamic/layout.md

```diff
@@ -0,0 +1,335 @@
+
+[!NOTE]: The order of the bytes of an integer larger than 1 byte is known as the endianness. All bytes representing an integer type are value bytes.
+
+[§](r[dynamic.layout.scalar.signed-repr]) Each signed integer type is represented the same as the equivalent value of the corresponding unsigned integer type, taken modulo `2^N`.
+
+[§](r[dynamic.layout.scalar.int-validity]) A given value of an integer type is valid if it was computed with only initialized bytes.
+
+[§](r[dynamic.layout.scalar.int-ptr-bytes]) Any pointer portion of a byte used in computing a value of an integer type is discarded.
+
+[§](r[dynamic.layout.scalar.int-size]) The special types `usize` and `isize` are integer types with the same target-dependent width.
+
+[!NOTE]: The width of `usize` and `isize` do not need to correspond to the width of any other integer type. If it does, then they have the same representation as that integer type.
+
+[§](r[dynamic.layout.scalar.char]) The type `char` has an underlying type of `u32`. 
+
+[§](r[dynamic.layout.scalar.char-validity]) A given value of type `char` is valid if the corresponding value of type `u32` is valid, and does not lie in the range `0xD800..0xE000` or the range `0x110000..`. 
+
+[§](r[dynamic.layout.scalar.bool]) The type `bool` has an underlying type of `u8`.
+
+[§](r[dynamic.layout.scalar.bool-repr]) The value `false` is represented the same as the value `0_u8`, and the value `true` is represented the same as the value `1_u8`
+
+[§](r[dynamic.layout.scalar.bool-validity]) A given value of type `bool` is valid if the corresponding value of type `u8` is valid, and the value lies in the range `0..2`.
+
+[§](r[dynamic.layout.scalar.float]) A floating point type with width `N` has an underlying type of `uN`.
+
+[§](r[dynamic.layout.scalar.float-repr]) A value of a floating-point has the representation corresponding to the value of the unsigned integer type computed from the value given by the appropriate interchange format from [IEEE 754](https://ieeexplore.ieee.org/document/5976968).
+
+[!NOTE]: The corresponding value may be called the bit representation of the floating-point value. 
+
+## Pointer Layout [§](r[dynamic.layout.pointer])
+
+[§](r[dynamic.layout.pointer.thin]) A pointer to a `Thin` type has a data portion only, which represents an address and a pointer tag. 
+
+[!NOTE]: Such pointers are called thin pointers. All `Sized` types are also `Thin`. More types may be considered `Thin` in future versions of the spec. The layout of `*const T` and `*mut T` are the same.
+
+[§](r[dynamic.layout.pointer.size-align]) The size and alignment of a thin pointer is the same as the size and alignment of `usize`. 
+
+[!NOTE] All thin pointers have the same
+
+[§](r[dynamic.layout.pointer.repr]) The address of a pointer value is the same as the value of type `usize` computed from the same bytes. The pointer tag of a pointer value is each pointer portion of each byte of the representation. 
```

Comment:
```suggestion
[§](r[dynamic.layout.pointer.repr]) The target of a pointer value is the same as the value of type `usize` computed from the same bytes. The pointer tag of a pointer value is each pointer portion of each byte of the representation. 
```

Reply:
address is a term used elsewhere in rust. For example, under the strict provenance API, there are functions to extract the address part that use the name address.

Reply:
> address is a term used elsewhere in rust. For example, under the strict provenance API, there are functions to extract the address part that use the name address.

Totally understand that. I was just trying to make some language to "show" the reader that the value (in this case) is what determines the target. 

Although, now that I reread it, I think that I understand the reason for the use of *address*: the reader will know (from other places in the standard) that the term identifies the thing being pointed to/at.

Sorry for the spam!